forte/data/ontology/code_generation_objects.py

```diff
@@ -856,17 +856,19 @@ def collect_parents(self, node_dict: Dict[str, Set[str]]):
 
         Args:
             node_dict: the nodes dictionary of nodes to collect parent nodes
-                for.
+                for. The entry represented by nodes in this dictionary do not store
```

Comment:
The entry node actually stores the type info of attributes according to your latest changes.

Reply:
So `node_dict` is another component just used in `collect_parents`, this does not store type information because we do [this](https://github.com/Pushkar-Bhuse/forte/blob/39b4de96d625ea3231b35432847bc2f467b27aaa/forte/data/ontology/code_generation_objects.py#L870). On the other hand, [here](https://github.com/Pushkar-Bhuse/forte/blob/39b4de96d625ea3231b35432847bc2f467b27aaa/forte/data/data_store.py#L1852) we use the node with its index and type information and I've mentioned it in the comments.